src/input-helper.ts

```diff
@@ -68,6 +76,10 @@ export function getInputs(): IGitSourceSettings {
         result.ref = `refs/heads/${result.ref}`
       }
     }
+
+    if (result.isGist && !result.ref && !result.commit) {
```

Comment:
added this back from https://github.com/actions/checkout/commit/00a3be89340a3ce8d704f82f44a5e7f9e3a84dfe

let me know if you'd like me to change `src/git-source-provider.ts` instead.